src/common/images/uploadImagesDialog.component.test.tsx

```diff
@@ -0,0 +1,160 @@
+import { fireEvent, screen, waitFor } from '@testing-library/react';
+import userEvent, { UserEvent } from '@testing-library/user-event';
+import { http, HttpResponse } from 'msw';
+import { act } from 'react';
+import { MockInstance } from 'vitest';
+import { server } from '../../mocks/server';
+import { renderComponentWithRouterProvider } from '../../testUtils';
+import UploadImagesDialog, {
+  UploadImagesDialogProps,
+} from './uploadImagesDialog.component';
+
+describe('Upload image dialog', () => {
+  let props: UploadImagesDialogProps;
+  let user: UserEvent;
+
+  let xhrPostSpy: MockInstance;
+
+  const onClose = vi.fn();
+
+  const createView = () => {
+    return renderComponentWithRouterProvider(<UploadImagesDialog {...props} />);
+  };
+
+  beforeEach(() => {
+    props = {
+      open: true,
+      onClose: onClose,
+      entityId: '1',
+    };
+    user = userEvent.setup();
+    xhrPostSpy = vi.spyOn(window.XMLHttpRequest.prototype, 'open');
+    global.URL.createObjectURL = vi.fn(() => 'mocked-url');
+    vi.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+    xhrPostSpy.mockRestore();
+  });
+
+  it('renders dialog correctly', async () => {
+    let baseElement;
+    await act(async () => {
+      baseElement = createView().baseElement;
+    });
+
+    expect(
+      screen.getByText('Files cannot be larger than', { exact: false })
+    ).toBeInTheDocument();
+    expect(baseElement).toMatchSnapshot();
+  });
+
+  it('calls onclose when close button is clicked', async () => {
+    createView();
+
+    await user.click(screen.getByLabelText('Close Modal'));
+
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  it('uploads an image with a title and description and verifies completion message', async () => {
+    // Render the component
+
+    createView();
+
+    const file1 = new File(['hello world'], 'image.png', {
+      type: 'image/png',
+    });
+
+    // Find the Uppy Dashboard's drop zone (it usually has a label like "Drop files here" or "Browse files")
+    const dropZone = screen.getByText('files cannot be larger than', {
+      exact: false,
+    });
+
+    // Create a drag-and-drop event for the file
+    Object.defineProperty(dropZone, 'files', {
+      value: [file1],
+    });
+
+    // Fire the drop event
```

Comment:
```suggestion
    // Fire the drop event
```
Could you remove these where they occur elsewhere too please (I think a few are in the attachments?). Same for other unnecessary ones.